### PR TITLE
Transcripts: speaker info

### PR DESCRIPTION
**Pull request by SergioEstevao**
| 📘 Part of: #1848  | 
|:---:|

This PR adds support for showing the speaker information on transcripts, if any is available.

| 1 | 2 |
| - | - |
| Simulator Screenshot - iPhone 15 - 2024-08-08 at 18 10 05 | Simulator Screenshot - iPhone 15 - 2024-08-08 at 16 06 56 |

## To test

- Start the app
- Ensure that you have the `transcripts` and `newShowNotesEndpoint` FF enabled
- Open this [episode](https://pca.st/8if3skl4) from Build Your SaaS that has transcript with speaker information
- Tap Play
- Open the full screen player
- Tap on transcripts
- Check that the transcript information is show correctly and you see the speakers info correctly:
  - Speaker names shows in its own line with a smaller font
  - Speaker names are not repeated in sequential paragraphs, new speaker information should only be show when a new speaker is talking
  - Speaker name have an empty line before then.
  -  Speaker name font size should change when accessibility font size is changed on Settings.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/TranscriptModel.swift [hunk 1]
```diff
@@ -58,4 +64,39 @@ struct TranscriptModel: Sendable {
...
+        case .vtt:
+            return regexMatch(input: cue.text, pattern: "<v (.+?)>", position: 1)
+        case .srt:
+            return regexMatch(input: cue.text, pattern: "^(.+?):", position: 1)
+        default:
+            return nil
+        }
+    }
+
+    private static func regexMatch(input: String, pattern: String, position: Int = 0) -> String? {
```
Comment (bjtitus) [inline podcasts/TranscriptModel.swift:82]: Would it make sense to add an iOS 16 implementation using [Regex](https://developer.apple.com/documentation/swift/regex) or wait to update later?

**Review by bjtitus — APPROVED** [review]
Looks good in testing! Just left one small note about iOS 16 API updates.

File: podcasts/TranscriptModel.swift (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/TranscriptModel.swift:82]: I personally prefer to keep this with a single implementation until we are are able to upgrade and use it on all devices.

**Comment by leandroalonso** [comment]
@SergioEstevao I did some general testing and it works as described. I shared a message in Slack about reducing spacing so we have more content on the screen but it's not a blocker for this task.

**Pull request merged**